x-pack/test/functional/services/ml/stack_management_jobs.ts

```diff
@@ -99,45 +97,38 @@ export function MachineLearningStackManagementJobsProvider(
     },
 
     async assertADJobRowSpaces(adJobId: string, expectedSpaces: string[]) {
-      await mlADJobTable.refreshJobList('stackMgmtJobList');
-      const rows = await mlADJobTable.parseJobTable('stackMgmtJobList');
-      const jobRow = rows.filter((row) => row.id === adJobId)[0];
-      expect(jobRow).to.have.property('spaces');
-      expect(jobRow.spaces!.sort()).to.eql(
+      await this.refreshList();
+      const spaces = await this.getSpacesFromTable();
```

Comment:
This `assertADJobRowSpaces` service method should validate the space for one particular job, but this looks like it's not taking the job id into account and instead checks against the collection of all the spaces from all table rows. This doesn't fail with the way it's is currently used, as the test filters the table first and in our cases, this really just brings back one row. But IMO we should rely on that fact and make this service method independent / really only check the spaces from one particular job row. And the same for `assertDFAJobRowSpaces` below.

Reply:
Updated in https://github.com/elastic/kibana/pull/134893/commits/fa39fc25ce8bd674304641dfade7854216def157